src/main/java/nl/tudelft/ewi/devhub/server/web/resources/ProjectAssignmentsResource.java

```diff
@@ -340,14 +342,28 @@ public Response processReview(@Context HttpServletRequest request,
                                   @PathParam("deliveryId") Long deliveryId,
                                   @FormParam("grade") String grade,
                                   @FormParam("commentary") String commentary,
-                                  @FormParam("state") Delivery.State state) throws UnauthorizedException, ApiError {
+                                  @FormParam("state") Delivery.State state) throws UnauthorizedException, ApiError, GradingException {
 
         if(!(currentUser.isAdmin() || currentUser.isAssisting(group.getCourse()))) {
             throw new UnauthorizedException();
         }
 
-        Double gradeValue = grade.isEmpty() ? null : Double.valueOf(grade);
         Delivery delivery = deliveries.find(group, deliveryId);
+        Assignment assignment = delivery.getAssignment();
+        GradingStrategy gradingStrategy = assignment.getGradingStrategy();
+
+        Double gradeValue;
+
+        if (assignment.isAssignmentHasRubrics()) {
+            // MissingRubricExceptions should not happen here as the review button
+            // is disabled until all characteristics have been filled in
```

Comment:
No, the server actually throws a `MissingRubricException` but I didn't go through the effort of implementing exception handling for that because it simply cannot occur through Devhub (hence the comment)